src/transformers/models/llava/modeling_llava.py

```diff
@@ -433,8 +434,14 @@ def forward(
                         device=attention_mask.device,
                     )
 
+                    # Ensuring indices are within bounds - and avoid CUDA index errors
+                    # See https://huggingface.co/llava-hf/llava-1.5-7b-hf/discussions/6 for more details
+                    valid_indices = non_attended_tokens < extended_attention_mask.shape[1]
+                    new_batch_index = batch_index[valid_indices]
```

Comment:
I see why this applies for `new_non_attended_tokens = non_attended_tokens[valid_indices]` but not for `new_batch_index` as `extended_attention_mask.shape[0]` can be different and so have a different set of valid indices

Reply:
That would cause a shape mismatch error if we only change the `non_attended_tokens`.
batch_index and non_attended_tokens come from this operation:
`batch_index, non_attended_tokens = torch.where(first_layer_past_key_value == 0)`

It points to which sample in batch and what token index. Pointing to row and column in `extended_attention_mask`.